java/core/src/java/org/apache/orc/impl/ConvertTreeReaderFactory.java

```diff
@@ -1411,7 +1411,7 @@ public void setConvertVectorElement(int elementNum) {
       long wholeSec = (long) Math.floor(seconds);
       timestampColVector.time[elementNum] = wholeSec * 1000;
       timestampColVector.nanos[elementNum] =
-          1_000_000 * (int) Math.round((seconds - wholeSec) * 1000);
+          Math.max(0, 1_000_000 * (int) Math.round((seconds - wholeSec) * 1000));
```

Comment:
@abstractdog , I am confused on whether this is the correct fix for the issue. Is this negative value because of long overflow? @omalley , @prasanthj , @t3rmin4t0r , what would be the expected behavior in this case for ORC? Setting timestamp to `null`? Using `Long.MAX_VALUE`?

Reply:
@jcamachor: yes, this was about long overflow, so I agree, it should be handled accordingly in my opinion, the next commit is about this:
https://github.com/apache/orc/pull/431/commits/9c2b9098597aa44ca2264e31e1b6ac8bd0659467
please take a look @omalley, @prasanthj , @t3rmin4t0r

Reply:
The overflow case should become null. I know it feels weird, because in programming we use "null" to mean "no value." In SQL it means "error value," which you can see when you accidentally write "col1 = null" and get no matches.

Reply:
thanks @omalley! so, do you mean that isNull[elementNum] = true is enough for the overflow case (without any other assignments in the timestamp col vector)?

Reply:
you need that as well as cv.noNulls = false